tests/unit/tasks/server/livereload-server-test.js

```diff
@@ -60,24 +60,22 @@ describe('livereload-server', function() {
           expect(!!subject.liveReloadServer).to.equal(false);
         });
     });
-    it('informs of error during startup with custom port', function(done) {
+    it('informs of error during startup with custom port', async function() {
       let preexistingServer = net.createServer();
       preexistingServer.listen(1337);
-      subject
-        .setupMiddleware({
+
+      try {
+        await subject.setupMiddleware({
           liveReload: true,
           liveReloadPort: 1337,
           liveReloadPrefix: '/',
           port: 4200,
-        })
-        .catch(function(reason) {
-          expect(reason).to.equal(
-            `Livereload failed on http://localhost:1337.  It is either in use or you do not have permission.`
-          );
-        })
-        .finally(function() {
-          preexistingServer.close(done);
         });
+      } catch (e) {
+        expect(e.message).to.include(`Livereload failed on 'http://localhost:1337', It may be in use.`);
+      } finally {
+        await new Promise(resolve => preexistingServer.close(resolve));
```

Comment:
I'm curious about whether there's there a reason you're not using `Promise.resolve` here?
```suggestion
        await Promise.resolve(preexistingServer.close(resolve));
```

Reply:
We need a promise that fulfills only when the callback passed to `prexistingServer.close` is invoked. 

Unfortunately the return value of `preexistingServer.close` is not meaningful when it comes to the above mentioned goal as `preexistingServer.close(cb) === preexistingServer` which when awaited fullfills on the next tick, regardless if the server has closed or not.


Two other concerns with the provided recommendation:

* `await Promise.resolve(value)` can be safely reduced to `await value` assuming `Promise` is the platform provided promise, which in this case it should be.
* the `resolve` argument passed to `close` is not defined, and will trigger a thrown `ReferenceError`.